metaserver/src/test/scala/tests/compiler/SquiggliesTest.scala

```diff
@@ -25,7 +34,7 @@ object SquiggliesTest extends CompilerSuite {
       val input = Input.VirtualFile(name, original)
       val doc = Semanticdbs.toSemanticdb(input, compiler)
       val PublishDiagnostics(_, diagnostics) :: Nil =
-        SquiggliesProvider.squigglies(doc, linter)
+        squiggliesProvider.squigglies(doc).runSyncMaybe.right.get
```

Comment:
Test helpers would be nice here so we didn't have to encode patterns in our tests that we wouldn't use in our code like .right.get.

For example:
```scala
  def assertNoDiffRight(task: Task[String], expected: String) = {
    val taskResult = task.runSyncMaybe
    taskResult match {
      case Left(cause) => fail("task failed...wtf?")
      case Right(value) => assertNoDiff(value, expected)
    }
  }

  def check(name: String, original: String, expected: String): Unit = {
    test(name) {
      val input = Input.VirtualFile(name, original)
      val doc = Semanticdbs.toSemanticdb(input, compiler)
      val obtained = squiggliesProvider.squigglies(doc)
        .map{ pubDiag =>
          val diags = pubDiag.flatMap(_.diagnostics)
          diags.map { d: =>
            val pos = input.toPosition(d.range)
            pos.formatMessage(d.severity.getOrElse(???).toString, d.message)
          }
        }

      assertNoDiffRight(obtained.map(_.mkString("\n")), expected)
    }
  }
```

Reply:
For me it's most important that errors don't get lost, `Try[T].get` is better than `Either[Err, T].right.get` in this regard. For Either I typically do `val Right(foo) = blah` instead of `right.get` for this exact reason, at least you get the left value on match error

Reply:
Agreed.  That’s why rightTest or something like it is nice, it lets you make sure you always fail a test when you don’t get the right you want.  An anti-pattern I have seen (but not in this codebase) is developers doing SomeMonad.map(aTest) and having it fail silently.